File: test/app/src/main/java/com/example/test/MainActivity.java
package com.example.test;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    // 버튼이 클릭된 횟수를 저장하는 변수를 선언한다.
    int clickCount = 0;

    /**
     * MainActivity 가 최초 실행될 때 실행된다.
     * 자바독 주석을 확인하려면 ctrl + q 키를 누르자
     * @param savedInstanceState
     */

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // Activity 의 UI를 R.layout.activity_main으로 한다.
        setContentView(R.layout.activity_main);

        // "프로그래밍을 시작해보자!" 메세지를 잠시 보여준다.
        Toast.makeText(getApplicationContext(), "프로그래밍을 시작해보자!", Toast.LENGTH_LONG).show();

        // 레이아웃에 button 이라는 ID로 선언된 뷰에 클릭 이벤트 리스너를 등록한다.
        findViewById(R.id.button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // 클릭 카운트를 1회 증가시킨다.
                clickCount = clickCount + 1;

                // 버튼이 클릭되면 클릭된 횟수를 증가하면서 토스트 메세지를 보여준다.
                Toast.makeText(getApplicationContext(), "clickCount: " + clickCount, Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: KotlinSample/app/src/main/java/com/example/kotlinsample/MainActivity.kt
package com.example.kotlinsample

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.* // findViewById() 함수를 사용하지 않도록 함

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // button1이 클릭된 경우의 이벤트 리스너 설정
        button1.setOnClickListener {
            // Intent로 BmiJavaActivity를 타겟으로 지정하고 startActivity로 실행
            startActivity(Intent(this@MainActivity, BmiJavaActivity::class.java))
        }

        // button2가 클릭된 경우의 이벤트 리스너 설정
        button2.setOnClickListener {
            // Intent로 BmiKotlinActivity를 타겟으로 지정하고 startActivity로 실행
            startActivity(Intent(this@MainActivity, BmiKotlinActivity::class.java))
        }

        // button3이 클릭된 경우의 이벤트 리스너 설정
        button3.setOnClickListener {
            // Intent로 VariableJavaActivity를 타겟으로 지정하고 startActivity로 실행
            startActivity(Intent(this@MainActivity, VariableJavaActivity::class.java))
        }

        // button2가 클릭된 경우의 이벤트 리스너 설정
        button4.setOnClickListener {
            // Intent로 VariableKotlinActivity를 타겟으로 지정하고 startActivity로 실행
            startActivity(Intent(this@MainActivity, VariableKotlinActivity::class.java))
        }

        // button5가 클릭된 경우의 이벤트 리스너 설정
        button5.setOnClickListener {
            // Intent로 ControlJavaActivity를 타겟으로 지정하고 startActivity로 실행
            startActivity(Intent(this@MainActivity, ControlJavaActivity::class.java))
        }

        // button6이 클릭된 경우의 이벤트 리스너 설정
        button6.setOnClickListener {
            // Intent로 ControlKotlinActivity를 타겟으로 지정하고 startActivity로 실행
            startActivity(Intent(this@MainActivity, ControlKotlinActivity::class.java))
        }

    }
}
